### DISCUSSION TITLE
[Bug Report / Feature Request] Sub-string Searching in the Collection

### DISCUSSION THREAD

**Discussion by ghost** [strawberrymusicplayer/strawberry#1263] [Ideas]
Good Evening GitHub Members

I have recently moved to Strawberry (from Clementine). I am running Strawberry 1.0.18 under Manjaro Linux.

A problem exists with searching for sub-strings in the "collection" search area. If, for example, I search for "etude" it will return the results that I expect. However, if I search for "tud" it will not return the correct results. This same problem is seen in Clementine.

I do not know if this a Linux-only problem. As I do not use MS Windows I cannot test it there.

It would be extremely useful to have sub-string searching. Even better, would be to have regexp-based searching.

I hope to see this included in a future release.

Regards
Stephen Meatheringham
Canberra, Australia

**Comment by jonaski** [comment]
I'm aware of the limitations, as far as I remember it has to do with the sqlite full text search which is used for the collection.
I plan to get rid of the SQL-based search for the collection entirely which means rewriting parts of the collection model and search, it should get rid of these limitations also.

**Reply by ghost** [reply]
Many thanks for the quick reply.
